domains/Links/Observers/LinkObserver.php

```diff
@@ -14,8 +16,10 @@ public function creating(Link $link)
             ->unapproved()
             ->count();
 
-        throw_if(
-            $pendingCount >= config('links.max_unapproved_links'),
+        throw_unless(
+            optional(Auth::user())->hasRole(AccountTypeEnum::EDITOR)
+            || optional(Auth::user())->isTrusted()
```

Comment:
At this layer, we should not create a dependency on HTTP layer (authentication). Seeing that this rule is for when the link is being submitted, we could add a policy to validate these requirements when the related controller's action is hit.

Reply:
Ok, but this moves to validation to a "higher" layer, meaning that if a Link is created via any other flow than the `LinksStoreController` this validation won't occur.
I'll move it.